File: Python/perm.py
def swap(a,b):
    return b,a

def perm(list,i,n):
    if i == n:
        for i in range(0,n+1):
            print(list[i],end ='')
        print()
        return
    j = i
    for j in range(n+1):
        list[i],list[j] = swap(list[i],list[j])
        perm(list,i + 1,n)
        list[i],list[j] = swap(list[i],list[j])

if __name__ == '__main__':
    list = ['a','b','c']
    perm(list,0,2)

File: C/perm.c
#include <stdio.h>
#include <stdlib.h>

void swap(char *a,char *b){
    char temp = *a;
    *a = *b;
    *b = temp;
}

void perm(char *list,int i,int n){
    int j;
    if(i == n){
        for(j = 0;j <= n;j++){
            printf("%c",list[j]);
        }
        printf("\n");
        return;
    }
    for(j = i;j <= n;j++){
        swap(&list[i],&list[j]);
        perm(list,i + 1,n);
        swap(&list[i],&list[j]);
    }
}

int main(void){
    char list[3] = {'a','b','c'};
    perm(list,0,2);
}